src/main/scala/mg7/pipeline.scala

```diff
@@ -0,0 +1,209 @@
+  case class AssignConfig(val size: Int) extends AnyAssignConfig with CommonConfigDefaults
+  case class  MergeConfig(val size: Int) extends AnyMergeConfig  with CommonConfigDefaults
+  case class  CountConfig(val size: Int) extends AnyCountConfig  with CommonConfigDefaults
+
+  /* You can override these values to customize configuration for each step */
+  val splitConfig:  AnySplitConfig  = SplitConfig(inputSamples.size)
+  val blastConfig:  AnyBlastConfig  = BlastConfig(inputSamples.size)
+  val assignConfig: AnyAssignConfig = AssignConfig(inputSamples.size)
+  val mergeConfig:  AnyMergeConfig  = MergeConfig(inputSamples.size)
+  val countConfig:  AnyCountConfig  = CountConfig(inputSamples.size)
+
+  // Boilerplate definitions that are derived from the ones above:
+
+  case object dataMappings {
+
+    lazy val split = inputSamples.toList.map { case (sampleId, readsS3Resource) =>
+
+      DataMapping(sampleId, splitDataProcessing(parameters))(
+        remoteInput = Map(
+          data.mergedReads -> readsS3Resource
+        ),
+        remoteOutput = Map(
+          data.fastaChunks -> S3Resource(outputS3Folder(sampleId, "split"))
+        )
+      )
+    }
+
+    private def instanceS3client(): S3 = S3.create(
+      new AWSCredentialsProviderChain(
+        new InstanceProfileCredentialsProvider(),
+        new ProfileCredentialsProvider()
+      )
+    )
+
+    private def listChunks(s3prefix: AnyS3Address): List[S3Object] = {
+      instanceS3client().listObjects(s3prefix.bucket, s3prefix.key)
+    }
+
+    // These prefixes will be used several times, so they factored in methods:
+    private def blastChunksS3Prefix(sampleId: String): S3Folder = outputS3Folder(sampleId, "blast") / "chunks" /
+    private def blastNoHitsS3Prefix(sampleId: String): S3Folder = outputS3Folder(sampleId, "blast") / "no-hits" /
+
+    // Here we generate tasks/data mappings for the blast loquat one per each S3 object generated by the split loquat
+    lazy val blast = split.flatMap { splitDM =>
+      val sampleId = splitDM.label
+
+      listChunks( splitDM.remoteOutput(data.fastaChunks).resource )
+        .zipWithIndex
+        .map { case (chunkS3Obj, n) =>
+
+          DataMapping(s"${sampleId}.${n}", blastDataProcessing(parameters))(
+            remoteInput = Map(
+              data.fastaChunk -> S3Resource(chunkS3Obj)
+            ),
+            remoteOutput = Map(
+              data.blastChunk  -> S3Resource(blastChunksS3Prefix(sampleId) / s"blast.${n}.csv"),
+              data.noHitsChunk -> S3Resource(blastNoHitsS3Prefix(sampleId) / s"no-hits.${n}.fa")
+            )
+          )
+        }
+    }
+
+    // These prefixes will be used several times, so they factored in methods:
+    private def lcaAssignS3Prefix(sampleId: String): S3Folder = outputS3Folder(sampleId, "assign") / "lca" /
+    private def bbhAssignS3Prefix(sampleId: String): S3Folder = outputS3Folder(sampleId, "assign") / "bbh" /
```

Comment:
I see `outputS3Folder(sampleId, "assign")` pattern everywhere; better add a step notion and build their output folder from that generically. That step type would help in other places too.